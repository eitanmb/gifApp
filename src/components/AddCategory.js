import React from 'react';
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleValue = (e) => {
        setInputValue( e.target.value );

        //para fines de las pruebas
        console.log('handleValue', inputValue);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        console.log('handleSubmit', inputValue);

        if(inputValue.trim().length > 2 ) {
            setCategories ( catsExistentes => [ inputValue, ...catsExistentes] );
            setInputValue('');
        }
    }
    
    return (
        <form onSubmit={ e => handleSubmit(e) }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                onChange= { (e) => handleValue(e) }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
