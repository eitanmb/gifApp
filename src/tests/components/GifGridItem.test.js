import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {

    const title = "Un titulo";
    const url = "https://localhos/imagen.jpg";

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe dibujar correctamente el componente', ()=> {
        expect( wrapper ).toMatchSnapshot();
    });


    test('debe tener un párrafo con el title ', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
        
    });

    test('debe tener un párrafo con el title ', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
        
    });

    test('debe tener una imagen con las props igual al src y alt ', () => {

        const img = wrapper.find('img');

        expect( img.props().src ).toBe( url );
        expect( img.props().alt ).toBe( title );
        
    });

    test('div debe contener la clase animate__fadeIn ', () => {

        const div = wrapper.find('div');
        const className = div.props().className;
       
        expect( className.includes('animate__fadeIn') ).toBe(true);
        
    })
    
    
    
});