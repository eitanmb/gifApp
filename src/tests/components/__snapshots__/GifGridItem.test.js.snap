// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe dibujar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhos/imagen.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
